hail/python/hail/matrixtable.py

```diff
@@ -224,16 +222,133 @@ def partition_hint(self, n: int) -> 'GroupedMatrixTable':
         return self
 
     @typecheck_method(named_exprs=expr_any)
-    def aggregate(self, **named_exprs) -> 'MatrixTable':
-        """Aggregate by group, used after :meth:`.MatrixTable.group_rows_by` or :meth:`.MatrixTable.group_cols_by`.
+    def aggregate_cols(self, **named_exprs) -> 'GroupedMatrixTable':
+        """Aggregate cols by group.
+
+        Examples
+        --------
+        Aggregate to a matrix with cohort as column keys, computing the mean height
+        per cohort as a new column field:
+
+        >>> dataset_result = (dataset.group_cols_by(dataset.cohort)
+        ...                          .aggregate_cols(mean_height = agg.mean(dataset.pheno.height))
+        ...                          .result())
+
+        Notes
+        -----
+        The aggregation scope includes all column fields and global fields.
+
+        See Also
+        --------
+        :meth:`.result`
+
+        Parameters
+        ----------
+        named_exprs : varargs of :class:`.Expression`
+            Aggregation expressions.
+
+        Returns
+        -------
+        :class:`.GroupedMatrixTable`
+        """
+
+        if self._row_keys:
+            raise NotImplementedError("GroupedMatrixTable is already grouped by rows. Cannot aggregate over cols.")
+
+        assert self._col_keys is not None
+
+        existing_fields = list(self._parent.globals.dtype)
+        existing_fields.extend(self._col_keys.keys())
+        existing_fields.extend(list(self._parent.row.dtype))
+        if self._entry_fields is not None:
+            existing_fields.extend(list(self._entry_fields.keys()))
+
+        new_fields = self._col_fields if self._col_fields is not None else {}
+        for k, e in named_exprs.items():
+            if k in existing_fields or k in new_fields:
+                raise ExpressionException(f"GroupedMatrixTable.aggregate_cols cannot assign duplicate field '{repr(k)}'")
```

Comment:
repr will insert its own single quotes, can you find/replace these? sorry I wasn't clear.

The reason to do this is if the field has a newline in it, this will get escaped correctly (and will be copy-pastable into a Python interpreter!)